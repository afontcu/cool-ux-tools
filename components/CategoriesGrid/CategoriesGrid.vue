<template>
  <div class="Categories-grid">
    <nuxt-link v-for="category in categories"
               :key="category.name"
               :to="category.url"
               class="Category u-padding-small">
      {{ category.name }}
    </nuxt-link>
  </div>
</template>

<script>
import { categories } from '~/data/data.json'

export default {
  name: 'CategoriesGrid',
  data () {
    return {
      categories,
    }
  },
}
</script>

<style lang="scss">
.Categories-grid {
  $size: 12vmax;

  display: grid;
  grid-template-columns: repeat(auto-fill, minmax($size, 1fr));
  grid-auto-rows: $size;
  grid-gap: $global-spacing-unit;
}

.Category {
  text-align: center;
  background-color: #f1f1f2;
  transition: all ease 0.2s;

  &:hover {
    background-color: #acc743;
  }
}
</style>
