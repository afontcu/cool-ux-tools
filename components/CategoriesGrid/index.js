import CategoriesGrid from './CategoriesGrid'

export default CategoriesGrid
