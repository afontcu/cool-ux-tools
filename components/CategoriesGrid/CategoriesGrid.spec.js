import CategoriesGrid from './index.js'
import { mount } from '@vue/test-utils'

describe('CategoriesGrid.vue', () => {
  it('is a component', () => {
    const wrapper = mount(CategoriesGrid, {
      stubs: ['nuxt-link'],
    })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders main node', () => {
    const wrapper = mount(CategoriesGrid, {
      stubs: ['nuxt-link'],
    })
    expect(wrapper.classes()).toContain('Categories-grid')
  })

  it('renders a snapshot', () => {
    const wrapper = mount(CategoriesGrid, {
      stubs: ['nuxt-link'],
    })
    expect(wrapper.html()).toMatchSnapshot('default')
  })
})
