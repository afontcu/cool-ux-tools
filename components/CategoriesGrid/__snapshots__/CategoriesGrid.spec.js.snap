// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesGrid.vue renders a snapshot: default 1`] = `

<div class="Categories-grid u-flex u-flex-wrap u-flex-justify-between">
</div>

`;
